src/check-catalog/check-catalog.service.ts

```diff
@@ -21,62 +36,58 @@ export class CheckCatalogService implements OnApplicationBootstrap {
 
   async load() {
     const results = await this.getViewData();
-    CheckCatalogService.checkCatalog = results.map((i) => {
-      const parts = i.resultMessage.split("[").filter((i) => i.includes("]"));
+    CheckCatalogService.checkCatalog = results.map((item: { resultMessage: string; checkTypeCode: any; checkNumber: any; resultCode: any; }) => {
+      const parts = item.resultMessage.split("[").filter((part) => part.includes("]"));
       return {
-        code: `${i.checkTypeCode}-${i.checkNumber}-${i.resultCode}`,
-        message: i.resultMessage,
-        plugins:
-          parts && parts.length > 0 ? parts.map((i) => i.split("]")[0]) : [],
+        code: `${item.checkTypeCode}-${item.checkNumber}-${item.resultCode}`,
+        message: item.resultMessage,
+        plugins: parts.map((part) => part.split("]")[0]),
       };
     });
   }
 
-  static formatMessage(message: string, values?: {}, plugins?: string[]) {
-    if (!plugins || (plugins && plugins.length === 0)) {
-      const parts = message.split("[").filter((i) => i.includes("]"));
-      plugins =
-        parts && parts.length > 0 ? parts.map((i) => i.split("]")[0]) : [];
-    }
+  static formatResultMessage(code: string, values: FormattedValues = {}): string {
+    const result = CheckCatalogService.checkCatalog.find((item) => item.code === code);
 
-    if (plugins && values) {
-      plugins.forEach((plugin: string) => {
-        let i = plugin;
-        const regex = /[-|_|\/]/g;
-
-        if (i.match(regex)) {
-          i = i.replace(regex, " ");
-        }
-
-        const parts = i.split(" ");
-        let fieldname = `${parts[0].charAt(0).toLowerCase()}${parts[0].slice(
-          1
-        )}`;
-
-        if (parts.length > 1) {
-          parts.forEach((p: string, index: number) => {
-            if (index === 0) {
-              fieldname = fieldname.toLowerCase();
-            } else {
-              fieldname += `${p.charAt(0).toUpperCase()}${p
-                .slice(1)
-                .toLowerCase()}`;
-            }
-          });
-        }
-
-        message = message.replace(`[${plugin}]`, `[${values[fieldname]}]`);
-      });
+    if (!result) {
+      return `[${code}]`;
     }
 
+    let message = `[${code}] - ${result.message}`;
+    const plugins = result.plugins;
+
+    message = CheckCatalogService.formatMessage(message, {
+      stackPipeId: values.stackPipeId ?? 'N/A',
+      orisCode: values.orisCode ?? 'N/A',
+      systemID: values.systemID ?? 'N/A',
+      unadjustedHrlyValue: values.unadjustedHrlyValue ?? 'N/A',
+      sampleMethod: values.sampleMethod ?? 'N/A',
+      ...values,
+    }, plugins);
```

Comment:
The `formatMessage` function already coalesces undefined properties in `values` to 'N/A':
```suggestion
    message = CheckCatalogService.formatMessage(message, values, plugins);
```